client/landing/gutenboarding/domain-suggestions/selectors.ts

```diff
@@ -0,0 +1,15 @@
+/**
+ * Internal dependencies
+ */
+import { DomainSuggestionQuery } from './types';
+import { State } from './reducer';
+import { getSerializedDomainsSuggestionsQuery } from 'state/domains/suggestions/utils';
+
+export const getState = ( state: State ) => state;
+export const getDomainSuggestions = ( state: State, queryObject: DomainSuggestionQuery ) => {
+	const serializedQuery = getSerializedDomainsSuggestionsQuery( queryObject );
+	if ( serializedQuery ) {
+		return state.domainSuggestions[ serializedQuery ];
+	}
+	return null;
+};
```

Comment:
This can be landed like this at the moment but we should consider as a follow up touching these 2 points:
- Having the custom serialisation function called both here and in the reducer doesn't look so good. We can change this to a generic one like [json-stable-stringify](https://www.npmjs.com/package/json-stable-stringify).
- Consumer shouldn't be required to fill in all the static query data. Instead we can normalize the query before resolving it with a consumer-facing selector like this:
```
const getDomainSuggestions = ( query: string, options: Options ) => {
  const queryObject = normalizeQuery( query, options );
  return __internalQueryDomainSelector( queryObject );
}
```

Reply:
> This can be landed like this at the moment but we should consider as a follow up touching these 2 points:
> 
> * Having the custom serialisation function called both here and in the reducer doesn't look so good. We can change this to a generic one like [json-stable-stringify](https://www.npmjs.com/package/json-stable-stringify).
> * Consumer shouldn't be required to fill in all the static query data. Instead we can normalize the query before resolving it with a consumer-facing selector like this:
> 
> ```
> const getDomainSuggestions = ( query: string, options: Options ) => {
>   const queryObject = normalizeQuery( query, options );
>   return __internalQueryDomainSelector( queryObject );
> }
> ```

Both points are valid; however, I chose to remain consistent with the implementation in `client/state/domains/suggestions/`:

> Anyway, I wasn't quite sure how/where to plug in the defaults; I could obviously just merge the user specified query into a `defaultQuery` object, but I decided that it's easier to skip that for now and plug it in once we need it. It's also how existing domain suggestion code handles things; those values are expected to be supplied by the user pretty much everywhere, so we might choose to do so in Gutenboarding as well.

(Quoting from PR desc.)